.github/actions/cmx-versions/index.js

```diff
@@ -0,0 +1,81 @@
+const core = require('@actions/core');
+const fetch = require('node-fetch');
+const semverCoerce = require('semver/functions/coerce');
+const semverMajor = require('semver/functions/major');
+const semverMinor = require('semver/functions/minor');
+const semverGt = require('semver/functions/gt');
+
+async function getClusterVersions() {
+    const url = 'https://api.replicated.com/vendor/v3/cluster/versions';
+    const apiToken = core.getInput('replicated-api-token') || process.env.REPLICATED_API_TOKEN;
+    const headers = {
+        Authorization: apiToken
+    };
+
+    let clusterVersions = [];
+    try {
+        const response = await fetch(url, {
+            method: 'GET',
+            headers,
+        });
+
+        if (response.status === 200) {
+            const payload = await response.json();
+            clusterVersions = payload['cluster-versions'];
+        } else {
+            throw new Error(`Request failed with status code ${response.status}`);
+        }
+    } catch (error) {
+        console.error(`Error: ${error.message}`);
+        core.setFailed(error.message);
+    }
+
+    // versions to test looks like this:
+    // [
+    //   {distribution: k3s, version: v1.24},
+    //   ...
+    // ]
+    const versionsToTest = [];
+
+    clusterVersions.forEach((distribution) => {
+        const distroName = distribution.short_name;
+
+        if (distroName === 'helmvm' || distroName === 'kurl') {
+            // excluding the embedded distributions
+            return;
+        }
+
+        if (distroName === 'aks') {
+            // excluding aks for now
+            // TODO: add aks back in when we support some sort of filtering
+            return;
+        }
```

Comment:
AKS has started to become unstable and we also have limited capacity.  Thinking we should instead run one of the other one-off tests on AKS for now. Thoughts?